__tests__/local.test.js

```diff
@@ -103,6 +103,19 @@ describe('local', () => {
         expect(result).toEqual('qrux');
       });
     });
+
+    it('should return null in callback and promise if data is not set', () => {
+      const spy = jest.fn();
+      testStore.getItem('bar', spy).then((promiseResult) => {
+
+        // validate callback
+        expect(spy).toHaveBeenCalledTimes(1);
+        expect(spy).toHaveBeenCalledWith(null, null); // first null for no error, second for ret
```

Comment:
Running this test gives me:

```
 FAIL  __tests__/local.test.js
  ● Console

    console.info node_modules/localforage/dist/localforage.js:2621
      Redefining LocalForage driver: webExtensionLocalStorage

  ● local › getItem › should return null in callback and promise if data is not set

    expect(jest.fn()).toHaveBeenCalledWith(expected)

    Expected mock function to have been called with:
      [null, null]
    But it was called with:
      [null]

      at __tests__/local.test.js:113:21
```

It seems the return value is not `null, null`, but simply `null`. Does that rhyme with your expectations?

Reply:
I need to re-check. Thanks. Quiet interesting that the test passed in my IDE.

Reply:
Okay, as I saw here:

https://github.com/localForage/localForage/blob/abe16c8a20c78ecff359647ede8e97f6102601b5/src/utils/executeCallback.js#L1

and

https://github.com/localForage/localForage/blob/abe16c8a20c78ecff359647ede8e97f6102601b5/typings/localforage.d.ts#L18

[null, null] should be the correct return value. As you can see in my last commit, I fixed this behaviour and add a check for sync-storage, so it should run now for both also. :smile: